### ISSUE TITLE
Schedulers documentation improvements

### ISSUE THREAD

**Issue by DavidMGross** [ReactiveX/reactivex.github.io#114]
(Copied over from https://github.com/ReactiveX/RxJava/issues/2955)

JensRantil:

Source: http://reactivex.io/documentation/scheduler.html

* Neither of the schedulers has a working link to corresponding JavaDoc.
* Schedulers#computation documentation doesn't state how many threads it runs. Whether it runs async or really anything. I had to dig into source code and finally ended up on https://github.com/ReactiveX/RxJava/pull/2580/files which states it defaults to number of processors threads.
* Schedulers#io is missing the same documentation; how many threads? Does it grow infinitely?

**Comment by DavidMGross** [comment]
See also: https://github.com/ReactiveX/RxJava/issues/2944 for another thing this documentation might address.

(I've fixed the first two of Jens's bullet points; the third is still a TBD)

**Comment by JensRantil** [comment]
@DavidMGross Just making sure, did you close this issue without fixing bullet 3?

**Comment by DavidMGross** [comment]
I closed it in ReactiveX/RxJava after moving it to ReactiveX/reactivex.github.io where the Schedulers documentation lives (http://reactivex.io/documentation/scheduler.html), where the issue remains open.

**Comment by JensRantil** [comment]
@DavidMGross Yes, but _this_ issue is closed. In your comment when you closed it, you states that you had only fixed two bullets out of three. Has the last bullet been fixed, too? Or did you create a separate issue for that bullet somewhere?

**Comment by JensRantil** [comment]
Meh, sorry. I misunderstood the layout of issue pages... I now see #2955 is "Closed". Not this one. Apologize.

**Comment by JensRantil** [comment]
Looks like this is where the default schedulers are defined:https://github.com/ReactiveX/RxJava/blob/86147542573004f4df84d2a2de83577cf62fe787/src/main/java/rx/schedulers/Schedulers.java#L47 So, IOScheduler defaults to `CachedThreadScheduler`.

**Comment by DavidMGross** [comment]
I don't know much about CachedThreadScheduler.  It's something like a new
thread scheduler but it maintains a cache of threads whose users may expire
and get kicked off the thread or something along those lines so that
threads may be reused.

On Tue, May 26, 2015 at 4:23 AM, Jens Rantil <notifications@github.com>
wrote:

> Looks like this is where the default schedulers are defined:
> https://github.com/ReactiveX/RxJava/blob/86147542573004f4df84d2a2de83577cf62fe787/src/main/java/rx/schedulers/Schedulers.java#L47
> So, IOScheduler defaults to CachedThreadScheduler.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ReactiveX/reactivex.github.io/issues/114#issuecomment-105493198>
> .
>



-- 
David M. Gross
PLP Consulting

**Comment by JensRantil** [comment]
I skimmed through it; Basically this is how I understood it to work:
 * On new subscriber: Find an existing idling thread. If no thread, start a new one.
 * Once a subscriber thread is done, it stays around to be reused by another subscriber. However, if no subscriber reuses the thread withing 60 seconds it is killed.

**Comment by DavidMGross** [comment]
Here's the improved documentation: http://reactivex.io/documentation/scheduler.html
